src/components/graph/graph.helper.js

```diff
@@ -469,7 +469,104 @@ function normalize(vector) {
   return norm === 0 ? vector : { x: vector.x / norm, y: vector.y / norm };
 }
 
-const SYMBOLS_WITH_OPTIMIZED_POSITIONING = new Set([CONST.SYMBOLS.CIRCLE]);
+/**
+ * Calculate the length of a vector, from the center of a rectangle,
+ * to one of it's edges.
+ *
+ * @param {Object.<string, number>} RectangleCoords, The coords of the left-top vertex, and the right-bottom vertex.
+ * @param {Object.<string, number>} VectorOriginCoords The center of the rectangle coords.
+ * @param {Object.<string, number>} directionVector a 2D vector with x and y components
+ */
+function calcRectangleVectorLengthFromCoords({ x1, y1, x2, y2 }, { px, py }, directionVector) {
+  const angle = Math.atan(directionVector.y / directionVector.x);
+
+  const vx = Math.cos(angle);
+  const vy = Math.sin(angle);
+
+  if (vx === 0) {
+    return x2 - x1;
+  } else if (vy === 0) {
+    return y2 - y1;
+  }
+
+  const leftEdge = (x1 - px) / vx;
+  const rightEdge = (x2 - px) / vx;
+
+  const topEdge = (y1 - py) / vy;
+  const bottomEdge = (y2 - py) / vy;
+
+  return Math.min(...[leftEdge, rightEdge, topEdge, bottomEdge].filter(edge => edge > 0));
+}
+
+/**
+ * Calculate a the vector length from the center of a circle to it's perimeter.
+ *
+ * @param {number} nodeSize The size of the circle, when no viewGenerator is specified, else the size of an edge of the viewGenerator square.
+ * @param {boolean} isCustomNode is viewGenerator specified.
+ * @returns {number}
+ */
+function calcCircleVectorLength(nodeSize, isCustomNode) {
+  // because this is a circle and A = pi * r^2
+  // we multiply by 0.95, because if we don't the link is not melting properly
+  let radiusLength = Math.sqrt(nodeSize / Math.PI);
+  if (isCustomNode) {
+    radiusLength = nodeSize / 10 / 2;
+  }
```

Comment:
We might get a small optimization here by avoiding the `.sqrt` call for custom nodes

```suggestion
  let radiusLength;
  if (isCustomNode) {
    radiusLength = nodeSize / 10 / 2;
  } else {
    radiusLength = Math.sqrt(nodeSize / Math.PI);
  }
```